gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonSchema.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.parquet;
+
+import org.apache.gobblin.source.extractor.schema.Schema;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import parquet.schema.Type;
+
+import static org.apache.gobblin.converter.parquet.JsonSchema.DataType.ENUM;
+import static org.apache.gobblin.converter.parquet.JsonSchema.DataType.RECORD;
+import static parquet.schema.Type.Repetition.OPTIONAL;
+import static parquet.schema.Type.Repetition.REQUIRED;
+
+
+/**
+ * Builds a {@link Schema} from jsonSchema represented using json string or JsonArray.
+ */
+public class JsonSchema extends Schema {
+  private static final String RECORD_FIELDS_KEY = "values";
+  private static final String DEFAULT_RECORD_COLUMN_NAME = "temp";
+  private static final String TYPE_KEY = "type";
+  private static final String ENUM_SYMBOLS_KEY = "symbols";
+  private static final String COLUMN_NAME_KEY = "columnName";
+  private static final String DATA_TYPE_KEY = "dataType";
+  private static final String COMMENT_KEY = "comment";
+  private static final String DEFAULT_VALUE_KEY = "defaultValue";
+  private static final String IS_NULLABLE_KEY = "isNullable";
+  private final DataType type;
+
+  public enum DataType {
+    STRING, INT, LONG, FLOAT, DOUBLE, BOOLEAN, ARRAY, ENUM, RECORD, MAP
+  }
+
+  public JsonSchema(JsonArray jsonArray) {
+    JsonObject temp = buildTempRecord(jsonArray);
+    setSuper(temp);
```

Comment:
Is calling `super.setXXX` is the same as `setXXX`? It can be just `setFromJsonObject(jsonObject)`?